src/istio/authn/context.proto

```diff
@@ -51,6 +51,22 @@ message X509Payload {
   // Identity carries by X509 certification. This is extracted from SAN field
   // for Istio provided certificate.
   string user = 1;
+
+  // The trust domain corresponds to the trust root of a system.
+  // Refer to
+  // [SPIFFE-ID](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain)
+  string trust_domain = 2;
```

Comment:
Can this ever be empty?

Reply:
The user facing config (in MeshConfig) can be empty. If it is empty, we use "cluster.local". We can use the same logic here.  Phillip, can you add a comment specifying the default value?

Reply:
The filter is not using trust domain information as of today so I guess it can be empty. If there is no aliases, I guess we don't need to convert this trust domain to the config either, so I think we can leave it as empti-able?

Reply:
It's not obvious how this information will be used in each case. Are they used to verify the x509 cert? If it is empty, what does this mean etc. 

As pilot will populate this, so it doesn't matter if it is empti-able or not. But from the implementation point of view, it is clearer (and more independent from pilot/istio logic) if the field is always set explicitly.

Reply:
Merged two fields to one and added some comment. PTAL. Thanks!